sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/azureml_image_context.py

```diff
@@ -70,7 +71,8 @@ def __init__(
 
         if yaml_code_directory_path:
             local_code_mount_path = str(yaml_code_directory_path)
-            docker_code_folder_name = Path(yaml_code_directory_path).name
+            # Set the directory containing scoring script as AML_APP_ROOT/working directory
+            docker_code_folder_name = Path(yaml_code_directory_path, os.path.dirname(Path(yaml_code_scoring_script_file_name))).name
```

Comment:
It is not immediately clear to me how these are different. Do you have an example of when they differ?

Reply:
It happens whenever code_configuration.scoring_script is referring to a different folder as the code_configuration.code.
For example, when `code_configuration.scoring_script: scoring/score.py`.

Reply:
Example, CodeConfiguration(code="D:/azureml-examples/sdk/python/endpoints/online/model-1/", scoring_script="onlinescoring/score.py"),

Initially, docker_code_folder_name = model-1/
After this PR, docker_code_folder_name = scoring/

Reply:
> It happens whenever code_configuration.scoring_script is referring to a different folder as the code_configuration.code.

I thought `code_configuration.scoring_script` has to be a path under `code_configuration.code`? I believe the current logic matches what managed online endpoint is doing. Is there a reason you expect `docker_code_folder_name` to be `onlinescoring/` in the above example?

Reply:
These changes were not correct. I have updated the logic.